bigquery/google/cloud/bigquery/table.py

```diff
@@ -1408,7 +1426,23 @@ def _to_dataframe_bqstorage_stream(
         # the end using manually-parsed schema.
         return pandas.concat(frames)[columns]
 
-    def _to_dataframe_bqstorage(self, bqstorage_client, dtypes):
+    def _process_progress_updates(self, progress_queue, progress_bar):
```

Comment:
Does this work well for large tables as the number of updates grows, since you're potentially walking it every _PROGRESS_INTERVAL seconds?

Reply:
I decreased `_PROGRESS_INTERVAL` to account for the fact that we get way too many updates in a second, but you're right that with large tables (many streams) this becomes worse.

Originally I tried having a constant loop of updates for `tqdm` but the 

Right now I handle this by very likely dropping updates when the queue fills up, meaning the progress bar is grossly inaccurate for large tables. Maybe what I should do instead is have 2 queues. 🤔 One queue used by the workers and a "sum" thread to add up and send to main thread every `_PROGRESS_INTERVAL` via a different queue.

Reply:
I suppose the problem is also bounded by the max dataframe users can fit in ram.  Our truly large tables aren't really going to be pulled into dataframes without excessive projection or filtering.  Perhaps this is a non issue for now.